gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -189,6 +206,28 @@ function roleselection.SaveLayers()
     end
 end
 
+---
+-- Initializes the player's role weights to be their minimum value.
+--
+-- @param Player ply
+-- @realm server
+function roleselection.InitializeRoleWeights(ply)
+    -- Initialize the weight table
+    local minWeight = roleselection.cv.ttt_role_derandomize_min_weight:GetInt()
+    local roleWeightTable = {}
+    local roleList = roles.GetList()
+
+    for i = 1, #roleList do
+        local roleData = roleList[i]
+
+        if not roleData.isAbstract then
+            roleWeightTable[roleData.index] = minWeight
+        end
+    end
+
+    ply:SetRoleWeightTable(roleWeightTable)
+end
+
```

Comment:
How about we do not initialize this?
Because it will already use the minWeight when none is found and the table is then being filled inplace when it would increment each entry anyway.

I'd also remove the accessor and directly work on the table, to remove unnecessary functions.
We can then also initialize the table on the player as an empty table.

I think that would just simplify the code.

Reply:
We can't *just* leave it uninitialized; that was actually what my initial version did. I realized though that we still probably want to count rounds before the player gets the role the first time, which means initializing it more eagerly than I originally had it. It might be possible to initialize in the `get_weight` functions when doing role dist, but I worry that it'll miss roles for players, and lead to incorrect counting.

Reply:
This is also related somewhat to the minimum adjustment problem, where the only viable way to adjust the minimum is to fully reset the tables. If/when this table can persist across map changes and server restarts, that'll become an even bigger problem, and we'll probably need a concommand to force-reset anyway.